apps/api/src/server/utils/published-fields-changed.js

```diff
@@ -0,0 +1,49 @@
+import { get, isEqual } from 'lodash-es';
+
+/** @typedef {import('@pins/applications.api').Schema.Case} Case */
+
+/**
+ * Checks whether the given column name is included when the case is published
+ *
+ * @param {Case} original
+ * @param {Case} updated
+ * @returns {boolean}
+ * */
+export const publishedFieldsHaveChanged = (original, updated) => {
```

Comment:
This looks like it'd work fine, but if we change the schema/payload we have to update this too.

What if instead we build an event with the before and after cases, and compare them (i.e. using `buildNsipProjectPayload`)? Could even just stringify and check if they're equal. That way it's tied into what we publish.

Reply:
Yeah good idea, I've made that change. We just need to make sure the `buildNsipProjectPayload` function only includes properties that should be in the event. There's some `@ts-ignore` dotted around that might allow extra properties to be included. I think it's right for now though.